openchs-server-api/src/main/java/org/openchs/service/ProgramEncounterService.java

```diff
@@ -44,4 +72,106 @@ public ProgramEncountersContract constructProgramEncounters(ProgramEncounter pro
         return  programEncountersContract;
     }
 
+    public ProgramEnrolment getAllEncountersForEnrolment(String uuid){
+        ProgramEnrolment programEnrolment = programEnrolmentRepository.findByUuid(uuid);
+        return programEnrolment;
+    }
+
+    public List<ProgramEncounter> constructVisitSchedules(ProgramEnrolment programEnrolment,String encounterTypeName){
+        return programEnrolment.getProgramEncounters().stream().filter(programEncounter -> programEncounter.getEncounterType().getOperationalEncounterTypeName().equals(encounterTypeName)).collect(Collectors.toList());
+    }
+
+    public void saveVisitSchedules(String uuid,List<VisitSchedule> visitSchedules){
+        for( VisitSchedule visitSchedule : visitSchedules){
+            try {
+                processVisitSchedule(uuid, visitSchedule);
+            }catch (Exception e){
+                RuleFailureLog ruleFailureLog = new RuleFailureLog();
+                ruleFailureLog.setFormId(uuid);
+                ruleFailureLog.setRuleType("Save : Visit Schedule Rule");
+                ruleFailureLog.setEntityId(uuid);
+                ruleFailureLog.setEntityType("Save : Program Encounter Save");
```

Comment:
Instead of giving a string for setEntityType, we should create and use the enum so it's always from a fixed set of values.